runtime/fetcher/fakes_test.go

```diff
@@ -0,0 +1,228 @@
+package fetcher
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"errors"
+	"io"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/taskcluster/taskcluster-worker/runtime/client"
+)
+
+type fakeContext struct {
+	context.Context
+	queue           client.Queue
+	m               sync.Mutex
+	progressReports []float64
+}
+
+func (c *fakeContext) Queue() client.Queue {
+	return c.queue
+}
+
+func (c *fakeContext) Progress(description string, percent float64) {
+	c.m.Lock()
+	defer c.m.Unlock()
+	debug("Progress: %s - %.02f %%", description, percent*100)
+	c.progressReports = append(c.progressReports, percent)
+}
+
+func (c *fakeContext) ProgressReports() []float64 {
+	c.m.Lock()
+	defer c.m.Unlock()
+	return c.progressReports
+}
+
+type fakeWriteReseter struct {
+	offset int64
+	buffer []byte
+}
+
+func (w *fakeWriteReseter) Write(p []byte) (int, error) {
+	offset := w.offset + int64(len(p))
+	if int64(len(w.buffer)) < offset {
+		w.buffer = append(w.buffer, make([]byte, offset-int64(len(w.buffer)))...)
+	}
+	copy(w.buffer[w.offset:], p)
+	w.offset = offset
+	return len(p), nil
```

Comment:
This can be summarized to:
```go
w.offset := w.offset + int64(len(p))
w.buffer = append(w.buffer, p)
return w.offset, nil
```

But I don't see why `offset` is necessary

Reply:
This is old code I just renamed it here... That should be clear from commits.

But I'm happy to adopt this :)  it's nicely spotted.